lib/routes/routeVeeam.js

```diff
@@ -67,8 +74,33 @@ function checkBucketAndKey(bucketName, objectKey, requestQueryParams, method, lo
         // Download relies on GETs calls with auth in query parameters, that can be
         // checked if 'X-Amz-Credential' is included.
         // Deletion requires that the tags of the object are returned.
-        if (requestQueryParams && Object.keys(requestQueryParams).length > 0
-            && !(method === 'GET' && (requestQueryParams['X-Amz-Credential'] || ('tagging' in requestQueryParams)))) {
+        const originalQuery = requestQueryParams || {};
+        const normalizedQueryKeys = new Map(
+            Object.keys(originalQuery).map(key => [key.toLowerCase(), key])
+        );
+        const isPresignedGet = method === 'GET'
+            && (normalizedQueryKeys.has('x-amz-credential')
+                || normalizedQueryKeys.has('tagging'));
+
+        if (!isPresignedGet && normalizedQueryKeys.size > 0) {
+            const hasUnsupportedKeys = Array.from(normalizedQueryKeys.keys())
+                .some(lowerKey => {
```

Comment:
this is very inefficient: instead of just going through the list one, you
- create a new map (converting & storing every header in lowercase)
- build yet another array of the keys
- iterate over every entry + explicitly check if presigned and/or x-id

```suggestion
let presigned = false;
for [key, value] in Object.entries(requestQueryParams || {}) {
    const normalizedKey = key.toLowerCase();
    if(normalizedKey == "x-id" && value !== apiToAction[method]) {
        return errorInstances.InvalidRequest
            .customizeDescription('The Veeam SOSAPI folder does not support this action.');
    } 
    if (normalizedKey == 'x-amz-credential' || normalizedKey == 'tagging') {
        presigned = true;
    }
    if (normalizedKey != 'x-amz-user-agent' && !normalizedKey.startsWith('x-amz-sdk-')) {
        return errorInstances.InvalidRequest
            .customizeDescription('The Veeam SOSAPI folder does not support this action.');
    }
}
```